### PR TITLE
feat(v2): add sticky footer

### PR DESCRIPTION
**Pull request by lex111**
## Motivation

Stick footer to bottom of page like as in v1.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yes.

## Test Plan

- Zoom out on any page or open page with small content (eg 404 page)

| Before   | After    |
| -------- | -------- |
| image| image |

### CODE REVIEW

**Comment by lex111** [comment]
Any updates? :smiley:

File: packages/docusaurus-theme-classic/src/theme/Layout/styles.css [hunk 1]
```diff
@@ -5,8 +5,23 @@
...
+  height: 100%;
+}
+
 body {
   margin: 0;
   padding-top: var(--ifm-navbar-height);
   transition: var(--ifm-transition-fast) ease color;
 }
+
+body > div {
```
Comment (yangshun) [inline packages/docusaurus-theme-classic/src/theme/Layout/styles.css]: It's better not to use tags directly (except for html/body)

Comment (yangshun) [reply packages/docusaurus-theme-classic/src/theme/Layout/styles.css]: Could we add these styles to `body` instead?

Comment (lex111) [reply packages/docusaurus-theme-classic/src/theme/Layout/styles.css]: > Could we add these styles to body instead?

Nope, we need to apply styles to the root element, i.e. id="__docusaurus", but it seemed like a bad use to me (this name can change, but here `body > div` applies just to the root element, which will always be one in the `body` tag.

The same goes for the Bootstrap approach.

Comment (lex111) [reply packages/docusaurus-theme-classic/src/theme/Layout/styles.css]: By the way, as for the Bootstrap approach - there the styles are also applied to the footer, which can be hidden (`noFooter`), then CSS rules for it needs to be defined in its component - Footer, not Layout. I want to say that the current approach is more integral - the change is only in Layout.

**Pull request merged**